tool/gocross/gocross-wrapper.sh

```diff
@@ -71,15 +60,38 @@ if [[ ! -d "$toolchain" ]]; then
             # Go uses the name "amd64".
             HOST_ARCH="amd64"
         fi
-
-        rm -rf "$toolchain" "$toolchain.extracted"
         curl -f -L -o "$toolchain.tar.gz" "https://github.com/tailscale/go/releases/download/build-${REV}/${HOST_OS}-${HOST_ARCH}.tar.gz"
         mkdir -p "$toolchain"
         (cd "$toolchain" && tar --strip-components=1 -xf "$toolchain.tar.gz")
         echo "$REV" >"$toolchain.extracted"
         rm -f "$toolchain.tar.gz"
-        ;;
-    esac
+
+        # Do some cleanup of old toolchains while we're here.
+        for hash in $(find "$HOME/.cache/tsgo" -type f -maxdepth 1 -name '*.extracted' -mtime 90 -exec basename {} \; | sed 's/.extracted$//'); do
```

Comment:
Fine as written, though a bit safer alternative (because not susceptible to expansion of spaces) would be
```suggestion
        find "$HOME/.cache/tsgo" -type f -maxdepth 1 -name '*.extracted' -mtime 90 -exec basename {} \; | sed 's/.extracted$//' | while read -r hash ; do
```

YMMV.

Reply:
Minor thought: could we do this in gocross instead? Being in Go, the facilities for dealing with locking and deciding what do delete might be nicer?

My general idea when writing this script was to do the minimum necessary to hoist ourselves into a go program, and then do more complex things from there with better language/library support. I'm not dead set on that if you feel differently, that's just where I was coming from when I first wrote this.

Reply:
So one complication is that Android has its own fork of this script (in its `tool/go`) that doesn't use gocross. We need to either port this over there or move it to using gocross, which might be more trouble than it's worth, as it doesn't need anything in gocross. So I was thinking of later either adding the don't-use-gross env option to this wrapper (so it can even avoid building gocross) or breaking the shell script in two halves: one that downloads the toolchain only and one that builds+runs gocross.  But I was going to talk to people more about that later. @raggi had some thoughts too.

But if we end up splitting it in half, we'd want the cleaning half to be in shell. (I'd definitely prefer to be writing Go, though)

And if I move it to Go, we have to decide how often to run it... on every gocross invocation seems overkill? Or does the shell script have to track a dirty bit of when it did a curl download and then later, after gocross is built, run some new gocross subcommand? Seemed maybe more convoluted. I wasn't sure.

So I think I'll leave it for now but happy to change it later.